deploy/table_facility.sql

```diff
@@ -0,0 +1,42 @@
+-- Deploy ggircs:table_facility to pg
+-- requires: schema_ggircs
+
+begin;
+
+create table ggircs.facility (
+    id                        integer primary key,
+    report_id                 integer references ggircs.report(id),
+    organisation_id           integer references ggircs.organisation(id),
+    parent_facility_id        integer references ggircs.facility(id) deferrable initially immediate ,
```

Comment:
what's unique about the parent that it needs to be set `deferrable initially immediate `?
Is this a best practice that we should be including in all our declarations?

Reply:
I needed to do that so the key constraints could be deferred since it's a self-reference to the table, otherwise, the keying was throwing an FK constraint error

Reply:
I recall that coming up, thanks for clarifying

Reply:
Would adopting `deferrable initially immediate` help make export functions faster since we'd run the constraint validation at the end of each transaction instead of on each insert? If yes then maybe we could get a feature opened for discussion?